rpc/src/server/shell_handler.rs

```diff
@@ -67,9 +67,42 @@ pub async fn valid_blocks(_: Request<Body>, _: Params, _: Query, _: RpcServiceEn
     empty()
 }
 
-pub async fn head_chain(_: Request<Body>, params: Params, _: Query, env: RpcServiceEnvironment) -> ServiceResult {
+pub async fn head_chain(_: Request<Body>, params: Params, query: Query, env: RpcServiceEnvironment) -> ServiceResult {
     let chain_id = parse_chain_id(params.get_str("chain_id").unwrap(), &env)?;
-    make_json_stream_response(base_services::get_current_head_monitor_header(chain_id, env.state())?.unwrap())
+    let protocol = query.get_str("next_protocol");
+    make_json_stream_response(base_services::get_current_head_monitor_header(&chain_id, &env, protocol.map(|s| s.to_string()))?.unwrap())
+}
+
+pub async fn mempool_monitor_operations(_: Request<Body>, params: Params, query: Query, env: RpcServiceEnvironment) -> ServiceResult {
+    let chain_id = parse_chain_id(params.get_str("chain_id").unwrap(), &env)?;
+
+    let applied = query.get_str("applied");
+    let branch_refused = query.get_str("branch_refused");
+    let branch_delayed = query.get_str("branch_delayed");
+    let refused = query.get_str("refused");
+
+    let mempool_query = MempoolOperationsQuery{
+        applied: applied == Some("yes"),
+        branch_refused: branch_refused == Some("yes"),
+        branch_delayed: branch_delayed == Some("yes"),
+        refused: refused == Some("yes"),
+    };
+
+    make_json_stream_response(base_services::get_operations_monitor(&chain_id, &env, Some(mempool_query))?.unwrap())
+}
+
+pub async fn blocks(_: Request<Body>, params: Params, query: Query, env: RpcServiceEnvironment) -> ServiceResult {
```

Comment:
@adonagy 
hmm, looks suspicious,
do you have stored some calls? I would like to see the whole rpc path with parameters